### ISSUE TITLE
`implicit` parameter not used in the Sinkhorn Divergence Hessians notebook?

### ISSUE THREAD

**Issue by gjhuizing** [ott-jax/ott#320]
Hi,

I'm interested in computing Hessians of OT costs, so I had a look at the Sinkhorn Divergence Hessians notebook.

Did I misunderstand something or is the `implicit` parameter in the `loss` function not used? Especially since the %time_it seems to be exactly the same in the unrolled version and in the implicit version.

Should the code not be as below?
```python
"implicit_diff": implicit_lib.ImplicitDiff(precondition_fun=lambda x: x) if implicit else None
```

Also, when should I use the default `precondition_fun = lambda x: eps*jnp.log(x)`, and when should I use `precondition_fun = lambda x: x`?

Finally, a very minor remark: at some point the text reads "Let’s parse the three lines in the call" but due to linting there are more than three lines. 

Thanks a lot!

**Comment by michalk8** [comment]
Hi @gjhuizing ,

thanks for spotting this! In fact, in this notebook, implicit diff. was always being used, in #321 I corrected this (+ also not setting `symmetric` flag when computing Sinkhorn between `x/x` or `y/y`.

> Also, when should I use the default precondition_fun = lambda x: eps*jnp.log(x), and when should I use precondition_fun = lambda x: x?

As far as I remember, there's some issue with `precondition_fun`, pinging @ersisimou since she knows more than I do.
